api/src/main/java/com/epam/esm/controller/GiftCertificateController.java

```diff
@@ -0,0 +1,98 @@
+package com.epam.esm.controller;
+
+
+import com.epam.esm.entity.GiftCertificate;
+import com.epam.esm.service.GiftCertificateService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+/**
+ * Gift certificates controller class.
+ *
+ * @author Yauheni Tstiov
+ */
+@RestController
+@RequestMapping("/certificates")
+public class GiftCertificateController {
+    private final GiftCertificateService giftCertificateService;
+
+    /**
+     * Init the gift certificates controller class.
+     */
+    @Autowired
+    public GiftCertificateController(GiftCertificateService giftCertificateService) {
+        this.giftCertificateService = giftCertificateService;
+    }
+
+    /**
+     * * Finds all gift certificates by param
+     *
+     * @param tagName                        the name of certificate to be found.
+     * @param partGiftCertificateName        the part of name of gift certificate to be found.
+     * @param partGiftCertificateDescription the part of gift description of certificate to be found.
+     * @param sortByName                     the sort param by gift certification name.
+     * @param sortByCreateDate               the sort param by create date.
+     * @param sortByLastUpdateDate           the sort param by create date.
+     * @return list with found items.
+     */
+    @GetMapping
+    public List<GiftCertificate> findGiftCertificates(@RequestParam(required = false) String tagName,
+                                                      @RequestParam(required = false) String partGiftCertificateName,
+                                                      @RequestParam(required = false) String partGiftCertificateDescription,
+                                                      @RequestParam(required = false) String sortByName,
+                                                      @RequestParam(required = false) String sortByCreateDate,
+                                                      @RequestParam(required = false) String sortByLastUpdateDate) {
+        return giftCertificateService.findGiftCertificateByIdWithTagsAndParams(tagName, partGiftCertificateName, partGiftCertificateDescription, sortByName, sortByCreateDate, sortByLastUpdateDate);
```

Comment:
long line, the max length is 160, please follow code style

Reply:
Done.